test/v1alpha1/route.go

```diff
@@ -102,13 +102,13 @@ func CheckRouteState(client *test.ServingAlphaClients, name string, inState func
 // IsRouteReady will check the status conditions of the route and return true if the route is
 // ready.
 func IsRouteReady(r *v1alpha1.Route) (bool, error) {
-	return r.Generation == r.Status.ObservedGeneration && r.Status.IsReady(), nil
+	return r.IsReady(), nil
 }
 
 // IsRouteNotReady will check the status conditions of the route and return true if the route is
 // not ready.
 func IsRouteNotReady(r *v1alpha1.Route) (bool, error) {
-	return !r.Status.IsReady(), nil
+	return r.Generation == r.Status.ObservedGeneration && !r.IsReady(), nil
```

Comment:
```suggestion
	return !r.IsReady(), nil
```

Reply:
The names of these are a bit misleading. The tests are actually looking for the resource to have been observed and failed (False). !IsReady() will do the wrong thing when the state is Unknown.

It looks like they are corrected in v1, but I'll making them uniform.

Reply:
Renamed to IsFooFailed to try to avoid that confusion